service/cloudconfig/template.go

```diff
@@ -6,6 +6,11 @@ const (
 	getKeyVaultSecretsFileName = "/opt/bin/get-keyvault-secrets"
 	getKeyVaultSecretsTemplate = `#!/bin/bash -e
 
+until $(curl --fail --output /dev/null http://localhost:50342/oauth2/token --data "resource=https://vault.azure.net" -H Metadata:true); do
```

Comment:
This script launches before the azure vault is created, so it fails by default.

By adding this (kind of ugly?) until condition, it just waits for the vault and the secrets to be ready before fetching them.

Reply:
Let's please split it to a separate PR so we can merge it quickly.